src/extremeweatherbench/config.py

```diff
@@ -186,3 +128,67 @@ def mapped_metadata_vars(self):
             for field in dataclasses.fields(self)
             if field.default in self.metadata_vars
         ]
+
+
+@dataclasses.dataclass
+class Config:
+    """High-level configuration for an ExtremeWeatherBench analysis run.
+
+    The Config class defines data ranges, output directories, and any custom
+    behavior that should used when running an end-to-end ExtremeWeatherBench
+    analysis. We prescribe sensible defaults for consistency, but these can be
+    extended as desired by the user.
+
+    Attributes:
+        event_types: A list of EventContainer event types to evaluate.
+            Base options currently include: HeatWave and Freeze,
+            Expanded options will include: SevereDay, AtmosphericRiver,
+            and TropicalCyclone with others to follow.
+        output_dir: A directory where outputs generated by the analysis should be saved.
+        forecast_dir: A directory where forecast data to be analyzed by EWB is stored.
+        cache_dir: A directory where intermediate data generated by an EWB analysis are stored.
+               Set to None to disable caching.
+        gridded_obs_path: A URI or filepath where a gridded observation dataset that can
+            be used for evaluation is stored. Defaults to ARCO-ERA5 on Google Cloud
+            Storage.
+        point_obs_path: A URI or filepath where a point observation dataset
+            that be used for evaluation is stored.
+        remote_protocol: The storage protocol which the forecast data is stored on.
+            Defaults to "s3".
+        forecast_preprocess: A function that preprocesses the forecast dataset into a
+            format expected by the evaluation metrics.
+        init_forecast_hour: The first forecast hour to include in the evaluation.
+            Defaults to 0.
+        temporal_resolution_hours: The resolution of the forecast data in hours.
+            Defaults to 6.
+        output_timesteps: The number of timesteps to include in the evaluation.
+            Defaults to 41.
+        gridded_obs_storage_options: A dictionary of cloud storage options for the gridded
+            observation dataset.
+        point_obs_storage_options: A dictionary of cloud storage options for the point
+            observation dataset.
+    """
+
+    event_types: List[events.EventContainer]
+    output_dir: Union[str, Path] = DEFAULT_OUTPUT_DIR
+    forecast_dir: Union[str, Path] = DEFAULT_FORECAST_DIR
+    cache_dir: Union[str, Path] = DEFAULT_CACHE_DIR
+    gridded_obs_path: Union[str, Path] = ARCO_ERA5_FULL_URI
+    point_obs_path: Union[str, Path] = DEFAULT_POINT_OBS_URI
```

Comment:
Because these could be remote URI's starting with something like `s3://` or `gs://`, I think the type should just be `str`. Elsewhere, you can check if they're remote URIs or not and then eagerly cast to `Path`.